### DISCUSSION TITLE
patch based pipeline and preprocess

### DISCUSSION THREAD

**Discussion by Chop1** [TorchIO-project/torchio#553] [Q&A] [answered]
From my understanding, all the `transform` are applied to the whole image, then the patchs are extracted.

So, the normalisation is not patch-based but image based.
Does that mean that I can perform all preprocessing and save it on disk, and only apply data-aumentation on the fly ?

**Comment by fepegar** [comment, marked as answer]
Hi, @Chop1. Sure, you can do that if you want and have the storage capacity for it. It will save you some training time.
